class.jetpack.php

```diff
@@ -3944,7 +3944,13 @@ function admin_page_load() {
 		if ( isset( $_GET['connect_url_redirect'] ) ) {
 			// User clicked in the iframe to link their accounts
 			if ( ! Jetpack::is_user_connected() ) {
-				$connect_url = $this->build_connect_url( true, false, 'iframe' );
+				$from = ! empty( $_GET['from'] ) ? $_GET['from'] : 'iframe';
```

Comment:
I'm not totally comfortable with only falling back to `iframe` without still really knowing what we need it for, and why we wouldn't need it in the JPO case. Think we can drop it altogether (also from its consumer)?

Reply:
I'd be hesitant to do so. At least not in this PR.